### PR TITLE
DashboardView - layout cleanup

### PR DESCRIPTION
**Pull request by kandl**
- make DashboardViewLayout generic to make it extendable with custom widgets
- support custom widget rendering in DashboardView
- add FluidLayoutFacade to work with the layout in a convenient way
- sizing & other implementation details exposed via DashboardViewLayout interfaces removed, moved to correct places
- added rowHeaderRenderer prop to FluidLayout for sane row header rendering

JIRA: RAIL-2869

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2041/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3673/ : SUCCESS in 14m 49s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2141/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3762/ : FAILURE in 14m 44s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2146/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3766/ : FAILURE in 11m 48s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2150/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3770/ : SUCCESS in 11m 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2151/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3771/ : SUCCESS in 12m 16s

File: libs/sdk-ui-ext/src/internal/dashboardEmbedding/DashboardLayout/DashboardLayoutContentRenderer.tsx [hunk 1]
```diff
@@ -1,75 +1,56 @@
 // (C) 2007-2020 GoodData Corporation
 import React, { CSSProperties } from "react";
 import cx from "classnames";
-import { IDashboardViewLayoutContentRenderer } from "./interfaces/dashboardLayoutComponents";
-import {
-    getDashboardLayoutContentHeightForRatioAndScreen,
-    getDashboardLayoutMinimumWidgetHeight,
-} from "./utils/sizing";
+import { IDashboardViewLayoutContentRenderProps } from "./interfaces/dashboardLayoutComponents";
+// import { isDashboardLayoutContent } from "./interfaces/dashboardLayout";
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/internal/dashboardEmbedding/DashboardLayout/DashboardLayoutContentRenderer.tsx:5]: Probably some leftover dev code

Comment (kandl) [reply libs/sdk-ui-ext/src/internal/dashboardEmbedding/DashboardLayout/DashboardLayoutContentRenderer.tsx]: Nice catch, thanks! Removed.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2157/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3775/ : SUCCESS in 11m 38s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR940

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/675/ : SUCCESS in 22s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/937/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3777/ : SUCCESS in 14m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/22/ : SUCCESS in 7m 37s (non-voting)

**Pull request merged**